### PR TITLE
Upgrade jetty version

### PR DESCRIPTION
**Pull request by david4096**
**Description**

Upgrade jetty version, forced transitive dependency while we wait for dropwizard

**Review Instructions**

Tests pass

**Issue**

https://ucsc-cgl.atlassian.net/browse/SEAB-6592

**Security and Privacy**

If there are any concerns that require extra attention from the security team, highlight them here and check the box when complete. 

- [x] Security and Privacy assessed

e.g. Does this change...
* Any user data we collect, or data location?
* Access control, authentication or authorization?
* Encryption features?

Please make sure that you've checked the following before submitting your pull request. Thanks!

- [x] Check that you pass the basic style checks and unit tests by running `mvn clean install`
- [x] Ensure that the PR targets the correct branch. Check the milestone or fix version of the ticket.
- [x] Follow the existing JPA patterns for queries, using named parameters, to avoid SQL injection
- [x] If you are changing dependencies, check the Snyk status check or the dashboard to ensure you are not introducing new high/critical vulnerabilities
- [x] Assume that inputs to the API can be malicious, and sanitize and/or check for Denial of Service type values, e.g., massive sizes
- [x] Do not serve user-uploaded binary images through the Dockstore API
- [x] Ensure that endpoints that only allow privileged access enforce that with the `@RolesAllowed` annotation
- [x] Do not create cookies, although this may change in the future
- [x] If this PR is for a user-facing feature, create and link a documentation ticket for this feature (usually in the same milestone as the linked issue). Style points if you create a documentation PR directly and link that instead.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -510,6 +510,7 @@
                                          keysmap.list and mvn dependency:tree. Weird -->
                                         <exclude>com.fasterxml.jackson.core:jackson-databind</exclude>
                                         <exclude>com.fasterxml.jackson.core:jackson-annotations</exclude>
+                                        <exclude>org.eclipse.jetty:jetty-webapp</exclude>
```
Comment (denis-yuen) [inline pom.xml:513]: should add separate code comment/explanation on this one,  exclusion can probably be removed in next dropwizard version

**Review by denis-yuen — APPROVED** [review]

**Review by svonworl — APPROVED** [review]
Per the CVE https://nvd.nist.gov/vuln/detail/CVE-2023-36478, 11.0.16 isn't vulnerable to the attack?  Probably ok/good to update the jetty version, anyway...

**Review by coverbeck — APPROVED** [review]
Approving, but curious about Steve's question.

**Comment by david4096** [comment]
Steve's right, up-to "excluding" is the word I missed. Closing we might as well wait for dropwizard in this case.

**Pull request closed without merging**